### PR TITLE
Add updates to CHANGELOG for optional dependencies

### PR DESCRIPTION
**Pull request by mauicv**

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,5 +1,31 @@
 # Change Log
 
+## [v0.10.0](https://github.com/SeldonIO/alibi-detect/tree/v0.10.0) (2022-06-01)
+[Full Changelog](https://github.com/SeldonIO/alibi-detect/compare/v0.10.0rc1...v0.10.0)
+
+### Added
+- **New feature** TensorFlow is now an optional dependency. Error messages for incorrect use of detectors that are dependent on missing optional dependencies have been improved to include installation instructions and be more informative.
+
+### Changed
+- The optional dependency work has resulted in some imports being reorganised. This doesn't affect legacy imports as long as the installed dependencies are the same. ([#538](https://github.com/SeldonIO/alibi-detect/pull/538)).
```
Comment (ascillitoe) [inline CHANGELOG.md:10]: Nitpick - Would it make more sense to say "The original imports will still work as long as the relevant optional dependencies are installed."? Wondering if the meaning of "legacy imports" is obvious, and if "installed dependencies are the same" is clear?

**Review by ascillitoe — APPROVED** [review]
Two minor comments but content LGTM!

File: CHANGELOG.md (hunk 1 above)
Comment (mauicv) [reply CHANGELOG.md:10]: Yep makes sense

File: CHANGELOG.md [hunk 2]
```diff
@@ -1,5 +1,31 @@
 # Change Log
 
+## [v0.10.0](https://github.com/SeldonIO/alibi-detect/tree/v0.10.0) (2022-06-01)
+[Full Changelog](https://github.com/SeldonIO/alibi-detect/compare/v0.10.0rc1...v0.10.0)
```
[the remark this answers is not in the document]
Comment (mauicv) [reply CHANGELOG.md:4]: Yeah, I wasn't really sure of the convention here. Happy to merge `v0.10.0rc1` and `v0.10.0`.

File: CHANGELOG.md [hunk 3]
```diff
@@ -1,9 +1,10 @@
 # Change Log
 
-## [v0.10.0rc1](https://github.com/SeldonIO/alibi-detect/tree/v0.10.0rc1) (2022-06-01)
-[Full Changelog](https://github.com/SeldonIO/alibi-detect/compare/v0.9.1...v0.10.0rc1)
+## [v0.10.0](https://github.com/SeldonIO/alibi-detect/tree/v0.10.0) (2022-06-01)
+[Full Changelog](https://github.com/SeldonIO/alibi-detect/compare/v0.10.0...v0.9.1)
 
 ### Added
+- **New feature** TensorFlow is now an optional dependency. Error messages for incorrect use of detectors that are dependent on missing optional dependencies have been improved to include installation instructions and be more informative.
```
Comment (jklaise) [inline CHANGELOG.md:7]: Nitpick: not sure I would put this under Added and especially New Feature as that should relate only to new library functionality. I would personally put this under Changed, e.g. see https://github.com/SeldonIO/alibi/releases. Understand if opinions differ though.

Comment (mauicv) [reply CHANGELOG.md]: No, I think I agree, I've moved it to `Changed`

**Comment by ascillitoe** [comment]
**Note to self:** Need to document this breaking change https://github.com/SeldonIO/alibi-detect/pull/564

**Pull request merged**